HPXMLtoOpenStudio/resources/hpxml_defaults.rb

```diff
@@ -76,6 +76,25 @@ def self.apply_building_construction(hpxml, cfa, nbeds)
       hpxml.building_construction.conditioned_building_volume = cfa * hpxml.building_construction.average_ceiling_height
     end
     hpxml.building_construction.number_of_bathrooms = Float(Waterheater.get_default_num_bathrooms(nbeds)).to_i if hpxml.building_construction.number_of_bathrooms.nil?
+    if hpxml.building_construction.has_flue_or_chimney.nil?
+      hpxml.building_construction.has_flue_or_chimney = false
+      hpxml.heating_systems.each do |heating_system|
+        if [HPXML::HVACTypeFurnace, HPXML::HVACTypeBoiler].include? heating_system.heating_system_type
+          next if heating_system.heating_efficiency_afue.nil?
+          next if heating_system.heating_efficiency_afue >= 0.89
+
+          hpxml.building_construction.has_flue_or_chimney = true
+        elsif [HPXML::HVACTypeFireplace].include? heating_system.heating_system_type
+          hpxml.building_construction.has_flue_or_chimney = true
+        end
+      end
+      hpxml.water_heating_systems.each do |water_heating_system|
+        next if water_heating_system.energy_factor.nil?
+        next if water_heating_system.energy_factor >= 0.63
```

Comment:
Need to handle water heaters with inputs of UEF; call Waterheater.calc_ef_from_uef for those.